src/apps/replication/lib/replica_learn.cpp

```diff
@@ -457,23 +464,48 @@ void replica::on_learn_reply(
     // local state is newer than learnee
     if (resp->last_committed_decree < _app->last_committed_decree())
     {
-        dwarn("%s: on_learn_reply[%016llx]: learnee = %s, learner state is newer than learnee (primary): %" PRId64 " vs %" PRId64,
+        dwarn(
+            "%s: on_learn_reply[%016llx]: learnee = %s, learner state is newer than learnee (primary): %" PRId64 " vs %" PRId64 ", create new app",
             name(), req->signature, resp->config.primary.to_string(),
             _app->last_committed_decree(),
             resp->last_committed_decree            
             );
 
+        // TODO(qinzuoyan):
+        // - we'd better backup the old data, which may be recovered in some way.
```

Comment:
the replication_app_base::reset_states() is called in simple_kv_service_impl::close(), where we need reset app states to make it as a new app.
Also, in rrdb::close(), we also need to call replication_app_base::reset_states().